src/views/virtualmachines/details/tabs/network/copmonents/modal/NetworkInterfaceFormFields/NetworkInterfaceNetworkSelect.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+
+import { NetworkAttachmentDefinitionModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console/models/NetworkAttachmentDefinitionModel';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { FormGroup, FormSelect, FormSelectOption } from '@patternfly/react-core';
+
+import { interfaceTypeTypes } from '../utils/constants';
+import { networkNameStartWithPod, podNetworkExists } from '../utils/helpers';
+
+type NetworkInterfaceNetworkSelectProps = {
+  vm: V1VirtualMachine;
+  networkName: string;
+  setNetworkName: React.Dispatch<React.SetStateAction<string>>;
+  setInterfaceType: React.Dispatch<React.SetStateAction<string>>;
+  setSubmitDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+  isEditing?: boolean | undefined;
+  editInitValueNetworkName?: string | undefined;
+};
+
+const NetworkInterfaceNetworkSelect: React.FC<NetworkInterfaceNetworkSelectProps> = ({
+  vm,
+  networkName,
+  setNetworkName,
+  setInterfaceType,
+  setSubmitDisabled,
+  isEditing,
+  editInitValueNetworkName,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [nads, loaded, error] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: NetworkAttachmentDefinitionModelGroupVersionKind,
+    isList: true,
+    namespace: vm?.metadata?.namespace,
+  });
+
+  const hasPodNetwork = React.useMemo(() => podNetworkExists(vm), [vm]);
+  const hasNads = React.useMemo(() => nads && nads.length > 0, [nads]);
+  const isPodNetworkingOptionExists =
+    !hasPodNetwork || (isEditing && networkNameStartWithPod(editInitValueNetworkName));
+
+  const canCreateNetworkInterface = React.useMemo(
+    () => hasNads || !hasPodNetwork,
+    [hasNads, hasPodNetwork],
+  );
+
+  const podNetworkingText = t('Pod Networking');
+
+  const handleChange = (value: string, event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    setNetworkName(value);
+    setInterfaceType(
+      value === podNetworkingText ? interfaceTypeTypes.MASQUERADE : interfaceTypeTypes.BRIDGE,
+    );
+  };
+
+  React.useEffect(() => {
```

Comment:
lol
this useEffect is not readable my dude.. 12 items in deps array
please re-write this, or write a big comment above it explaining what it does

Reply:
lol,
changed it a bit and also added comments so it could be a bit more readable